hoomd/mpcd/ParticleData.cc

```diff
@@ -416,6 +457,119 @@ void mpcd::ParticleData::initializeRandom(unsigned int N, const BoxDim& local_bo
         }
     }
 
+/*!
+ * \param density Density of particles inside sphere
+ * \param R Radius of sphere 
+ * \param local_box Local simulation box
+ * \param kT Temperature (in energy units)
+ * \param seed Random seed
+ * \param ndimensions Dimensionality
+ */
+void mpcd::ParticleData::initializeRandomSphere(Scalar density, Scalar R, const BoxDim& local_box, Scalar kT, unsigned int seed, unsigned int ndimensions)
+    {
+    // only one particle type is supported for this construction method
+    m_type_mapping.clear();
+    m_type_mapping.push_back("A");
+    // default particle mass is 1.0
+    m_mass = Scalar(1.0);
+
+    // throw runtime error if radius is negative or zero
+    if (R <= Scalar(0.0))
+        {
+        throw std::runtime_error("Radius of sphere can't be negative or zero");
+        }
+
+    // Calculating Ndraw - number of particles that should be inside the local box according to density
+    unsigned int Ndraw = std::round(density*local_box.getVolume((ndimensions == 2)));
+
+    // Allocating an Array for storing the random drawn positions
+    std::vector<Scalar4> pos_draw;
+    pos_draw.reserve(Ndraw);
+
+    // center of box
+    const Scalar3 lo = local_box.getLo();
+    const Scalar3 hi = local_box.getHi();
+
+    // random number generator
+    std::mt19937 mt(seed);
+    std::uniform_real_distribution<Scalar> pos_x(lo.x, hi.x);
+    std::uniform_real_distribution<Scalar> pos_y(lo.y, hi.y);
+    std::uniform_real_distribution<Scalar> pos_z(lo.z, hi.z);
+    std::normal_distribution<Scalar> vel(0.0, fast::sqrt(kT / m_mass));
+
+    // generating Ndraw random positions and keeping them if they lie inside sphere
+    for (unsigned int i=0; i < Ndraw; ++i)
+        {
+        const Scalar3 pos = make_scalar3(pos_x(mt), pos_y(mt), (ndimensions == 3) ? pos_z(mt) : Scalar(0.0));
+        const Scalar r = slow::sqrt(dot(pos, pos));
+        if (r < R)
```

Comment:
I missed this the first time, but use r^2 < R^2 here, it is faster:
```suggestion
    // generating Ndraw random positions and keeping them if they lie inside sphere
    const double Rsq = R*R;
    for (unsigned int i=0; i < Ndraw; ++i)
        {
        const Scalar3 pos = make_scalar3(pos_x(mt), pos_y(mt), (ndimensions == 3) ? pos_z(mt) : Scalar(0.0));
        const Scalar rsq = dot(pos, pos);
        if (r < Rsq)
```